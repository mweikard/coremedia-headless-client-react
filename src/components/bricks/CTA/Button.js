// @flow
import { Button } from '../../basic/Button/';
import { getColors } from '../../styles/utils';
import withStyles from '../../styles/withStyles';

const styles = (theme, props) => ({
  marginTop: '1rem',
  color: getColors(theme.palette, props.color).text,
});

export default withStyles(styles, 'CTA')(Button);
