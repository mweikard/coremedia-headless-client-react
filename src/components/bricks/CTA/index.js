// @flow
import React from 'react';
import PropTypes from 'prop-types';

import Button from './Button';

type Props = {
  label: string,
  color?: string,
};

const CTABrick = ({ label, color }: Props) => <Button color={color}>{label}</Button>;

CTABrick.propTypes = {
  label: PropTypes.string.isRequired,
  color: PropTypes.string,
};

export default CTABrick;
