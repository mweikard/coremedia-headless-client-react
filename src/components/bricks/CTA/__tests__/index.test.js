// @flow
import React from 'react';

import CTA from '..';

describe('CTA Component', () => {
  it('should render CTA using Button', () => {
    const wrapper = shallow(<CTA label="Label" />);
    expect(wrapper).toMatchSnapshot();
  });
});
