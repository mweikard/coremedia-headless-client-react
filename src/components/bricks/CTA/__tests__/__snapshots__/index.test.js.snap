// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTA Component should render CTA using Button 1`] = `
<styled.button
  theme={
    Object {
      "typography": Object {
        "button": Object {
          "color": "#000",
          "fontFamily": "Helvetica, Arial, sans-serif",
          "fontSize": "1rem",
          "fontWeight": 500,
          "textTransform": "none",
        },
      },
    }
  }
>
  Label
</styled.button>
`;
